artemis/src/main/java/tech/pegasys/artemis/services/beaconchain/StateTransition.java

```diff
@@ -42,7 +42,7 @@ public void initiate(BeaconState state, BeaconBlock block){
 
 
         // per-slot processing
-        slotProcessor(state);
+        slotProcessor(state, block);
```

Comment:
I believe the `block` being passed here to the `slotProcessor` is a new block object, not the `previousBlock`. So we still need to access the `previousBlock` from somewhere else to use it in `updateRecentBlockHashes`.